src/Centeva.ObjectStorage.Azure.FileShare/AzureFileShareConnectionFactory.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Azure.Core;
+using Azure.Identity;
+
+using Centeva.ObjectStorage.Connections;
+
+namespace Centeva.ObjectStorage.Azure.FileShare;
+
+public class AzureFileShareConnectionFactory : IConnectionFactory
+{
+    private const string ProviderName = "azue.fileshare";
+    private const string AccountName = "accountName";
+    private const string AccountKey = "accountKey";
+    private const string ClientId = "clientId";
+    private const string Endpoint = "endpoint";
+    private const string Container = "container";
+
+    public IObjectStorage? CreateConnection(ObjectStorageConnectionString connectionString)
+    {
+        if (connectionString.ProviderName != ProviderName)
+            return null;
+
+        var container = connectionString.GetRequired(Container);
+        var accountName = connectionString.GetRequired(AccountName);
+        var accountKey = (connectionString.Get(AccountKey) ?? "").Replace(' ', '+');
+        var clientId = connectionString.Get(ClientId);
+        var endpoint = connectionString.Get(Endpoint);
+
+        // If we have an account key, we use shared key authentication
+        if (accountKey is not null and not "")
```

Comment:
I like how you added support for Managed Identity here. I think it is a much more natural way to do auth as we containerize these applications.